### PR TITLE
fix: stop recording loader on error

### PR DESCRIPTION
**Pull request by amar-1995**

### CODE REVIEW

File: packages/roomkit-react/src/Prebuilt/components/MoreSettings/SplitComponents/MwebOptions.tsx [hunk 1]
```diff
@@ -102,7 +103,13 @@ export const MwebOptions = ({
   const isLandscapeHLSStream = useLandscapeHLSStream();
   const toggleVB = useSidepaneToggle(SIDE_PANE_OPTIONS.VB);
   const isLocalVideoEnabled = useHMSStore(selectIsLocalVideoEnabled);
-
+  const recordingState = useHMSStore(selectRecordingState);
+  useEffect(() => {
```
Comment (raviteja83) [inline packages/roomkit-react/src/Prebuilt/components/MoreSettings/SplitComponents/MwebOptions.tsx:107]: seems like this can be a custom hook.

File: packages/roomkit-react/src/Prebuilt/components/MoreSettings/SplitComponents/MwebOptions.tsx [hunk 2]
```diff
@@ -89,7 +88,7 @@ export const MwebOptions = ({
   const [openSettingsSheet, setOpenSettingsSheet] = useState(false);
   const [showEmojiCard, setShowEmojiCard] = useState(false);
   const [showRecordingOn, setShowRecordingOn] = useState(false);
-  const [isRecordingLoading, setIsRecordingLoading] = useState(false);
```
Comment (raviteja83) [inline packages/roomkit-react/src/Prebuilt/components/MoreSettings/SplitComponents/MwebOptions.tsx:91]: why is this removed? no longer needed?

Comment (amar-1995) [reply packages/roomkit-react/src/Prebuilt/components/MoreSettings/SplitComponents/MwebOptions.tsx:91]: no, recordingStarted can be used

File: packages/roomkit-react/src/Prebuilt/common/hooks.ts [hunk 3]
```diff
@@ -147,3 +154,68 @@ export const useKeyboardHandler = (isPaused: boolean, hlsPlayer: HMSHLSPlayer) =
 
   return handleKeyEvent;
 };
+export interface RTMPRecordingResolution {
+  width: number;
+  height: number;
+}
+export const useIsRecordingStartErroredOut = () => {
```
Comment (raviteja83) [inline packages/roomkit-react/src/Prebuilt/common/hooks.ts:161]: rename to useRecordingHandler

File: packages/roomkit-react/src/Prebuilt/common/hooks.ts [hunk 4]
```diff
@@ -147,3 +154,68 @@ export const useKeyboardHandler = (isPaused: boolean, hlsPlayer: HMSHLSPlayer) =
...
+  const startRecording = useCallback(
+    async (resolution: RTMPRecordingResolution | null = null) => {
+      try {
+        setRecordingState(true);
+        setIsRecordingLoading(true);
+        await hmsActions.startRTMPOrRecording({
+          resolution: getResolution(resolution),
+          record: true,
+        });
+        setIsRecordingLoading(false);
```
Comment (raviteja83) [inline packages/roomkit-react/src/Prebuilt/common/hooks.ts:180]: this can be added once after the try catch

**Review by raviteja83 — APPROVED** [review]
